var/spack/repos/builtin/packages/r/package.py

```diff
@@ -82,35 +83,46 @@ class R(Package):
     depends_on('pcre')
     depends_on('jdk')
 
-    patch('zlib.patch', when='@:3.3.1')
+    patch('zlib.patch', when='@:3.3.2')
```

Comment:
Yep, see [here](https://github.com/LLNL/spack/pull/2864#issuecomment-273562042) for my tests with 3.3.2 without your patch. I checked that your patch successfully applies to 3.3.2, but I didn't build it all the way yet.